develop/tutorials/articles/192-item-selector/02-creating-custom-entites-for-the-item-selector.markdown

```diff
@@ -1,151 +1,131 @@
+configured for? No problem. You can create a new entity.
+
+This tutorial explains how to create a new entity for the Item Selector.
+
+Go ahead and get started.
+
+## Creating Item Selector Criterion [](id=creating-item-selector-criterion)
+
+First, you must create a new criterion for your entity. Follow these steps to 
+create an Item Selector criterion:
+
+1.  Create a class that extends `BaseItemSelectorCriterion`.
+
+    The example below creates a class called `TaskItemSelectorCriterion`:
+
+        public class TasksItemSelectorCriterion extends 
+        BaseItemSelectorCriterion {
+        
+        }
+        
+    This class acts as an identifier for Item Selector.
+
+    Note that you can use this class to pass information to the view if
+    needed. For example, the [JournalItemSelectorCriterion](https://github.com/liferay/liferay-portal/blob/586f66c629b559e79c744559751ecb960218fe0b/modules/apps/web-experience/journal/journal-item-selector-api/src/main/java/com/liferay/journal/item/selector/criterion/JournalItemSelectorCriterion.java)
+    passes information about the primary key so the view can use it:
+
+        public class JournalItemSelectorCriterion extends 
+        BaseItemSelectorCriterion {
+        
+                public JournalItemSelectorCriterion() {
+                }
+        
+                public JournalItemSelectorCriterion(long resourcePrimKey) {
+                        _resourcePrimKey = resourcePrimKey;
+                }
+        
+                public long getResourcePrimKey() {
+                        return _resourcePrimKey;
+                }
+        
+                public void setResourcePrimKey(long resourcePrimKey) {
+                        _resourcePrimKey = resourcePrimKey;
+                }
+        
+                private long _resourcePrimKey;
+        
+        }
+
+    +$$$
+
+    **Note:** Criterion fields should be serializable and should expose a 
+    public empty constructor(as shown above).
+    
+    $$$
+
+2.  Create an OSGi component class that implements 
+    `BaseItemSelectorCriterionHandler`. Each criterion requires a criterion 
+    handler, which is responsible for obtaining the proper selection view.
+
+    The example below creates a criterion handler for the 
+    `TaskItemSelectorCriterion`:
+
+        @Component(service = ItemSelectorCriterionHandler.class)
+        public class TaskItemSelectorCriterionHandler extends 
+        BaseItemSelectorCriterionHandler<TaskItemSelectorCriterion> {
+
+            public Class <TaskItemSelectorCriterion> 
+            getItemSelectorCriterionClass() {
+                return TasksItemSelectorCriterionHandler.class;
+            }
+            
+            @Activate
+            @Override
+            protected void activate(BundleContext bundleContext) {
+                    super.activate(bundleContext);
+            }
+    
+        }
+        
+    The @Activate and @Override tokens are required to activate this OSGi 
```

Comment:
Hey @mwilliams2014 I think that you don't even need to override the method. As long as it's available in the Base class and you have the `-dsannotations-options:inherit` that should do it, but it would be great if you can double check it just to be 100% sure.